### DISCUSSION TITLE
Choose a specific warehouse to ship from?

### DISCUSSION THREAD

**Discussion by LeOndaz** [saleor/saleor#9247] [Q&A]
What do you think? is there any possible way to choose a specific warehouse to ship an order from?

**Comment by karolkielecki** [comment]
Hi @LeOndaz, do you mean you would like to allow a customer to choose a specific warehouse?

**Reply by mszuchman** [reply]
Hi @karolkielecki, I am working with @LeOndaz so I am jumping in to give you more context. We are building a hyperlocal experience where we have multiple warehouses deployed in a city and for each warehouse we define a delivery area. Depending on the location of the customer, we know what are the warehouses that can deliver to the customer; We ask location right after the customer opens the app

Knowing the warehouse, we also know the selection and inventory available. The limitation that we found is at the time of placing the order, we need to ensure the stock is deducted on the right warehouse

**Reply by karolkielecki** [reply]
Hey @mszuchman, I think you'll have to fork the code and modify it to your needs. But let me discuss it with the team. Maybe there is another way of doing it.

**Reply by mszuchman** [reply]
perfect, let me know the outcome after discussing with the team

**Comment by peteror** [comment]
Hi @mszuchman - I'm working on something similar. Our approach is to do the deduction from the warehouse inventory on fulfilment; this should be fairly soon after the customer makes the order. (we build a separate app for the warehouse people that uses the fulfilment API) 
In theory, you could do this without forking the code, just register an app to the 'order created' webhook and do the fulfilment based on that. (but you need to know the warehouse, we use a medatada field in each order for this)

**Reply by karolkielecki** [reply]
@mszuchman I think the solution described above by @peteror would work and makes sense. But there can be a potential small issue--Saleor chooses a warehouse from which products will be allocated during checkout.

**Reply by peteror** [reply]
hi @karolkielecki  - I did not know that Saleor chooses a warehouse at checkout. Does this happen when there are multiple warehouses in the same shipping zone? 
Currently, we define a unique shipping zone for each warehouse (and a separate shipping method too, a bit messy, but we don't start with a ton of warehouses and we can programmatically generate it later if needed)

**Reply by mszuchman** [reply]
Thanks @peteror! but as @karolkielecki mentioned it won't work for me as I have multiple warehouses in the same city, and during the checkout the inventory is allocated

**Reply by peteror** [reply]
Just like you, we have multiple warehouses in the same city. We assign the warehouse (and shipping zone) based on the customer's location when the app starts. The only drawback I found so far is you need to create a shipping method for each shipping zone.

**Reply by peteror** [reply]
Just did some testing - I have defined separate shipping zones and shipping methods and mapped them 1:1 to warehouses. No matter which shipping zone I choose, the allocation goes to the first warehouse in the list. @karolkielecki  - is this a bug or a feature?

**Reply by karolkielecki** [reply]
@peteror Let me check with the team.

**Reply by karolkielecki** [reply]
What's your version (Saleor Core and Dashboard)?

**Reply by peteror** [reply]
@karolkielecki  - thank you. 
dashboard: v3.0.0-b.11
core: dev (deployed from the Main branch about a month ago)

**Reply by karolkielecki** [reply]
Let me clarify it `I have defined separate shipping zones and shipping methods and mapped them 1:1 to warehouses.` - Do you have the same warehouse or a different one to each shipping zone?

**Reply by peteror** [reply]
I have created different warehouses, one for each shipping zone.
